client/src/components/dataset/sidebars/ChangeDataType.jsx

```diff
@@ -114,51 +115,39 @@ export default class ChangeDataType extends Component {
   componentWillMount() {
     const { column, newColumnType } = this.props;
     this.setState({
-      op: 'core/change-datatype',
-      args: {
-        columnName: column.columnName,
-        newType: newColumnType,
-        defaultValue: null,
-        parseFormat: 'YYYY-MM-DD',
-      },
-      onError: 'default-value',
+      transformation: Immutable.fromJS({
+        op: 'core/change-datatype',
+        args: {
+          columnName: column.get('columnName'),
+          newType: newColumnType,
+          defaultValue: null,
+          parseFormat: 'YYYY-MM-DD',
+        },
+        onError: 'default-value',
+      }),
       errorStrategy: 'empty-cell',
     });
   }
 
-
   mergeArgs(args) {
-    this.setState({ args: Object.assign({}, this.state.args, args) });
+    const transformation = this.state.transformation;
+    this.setState({ transformation: transformation.mergeIn(['args'], args) });
   }
 
   handleChangeErrorStrategy(errorStrategy) {
-    switch (errorStrategy) {
-      case 'emptyCell': {
-        const args = Object.assign({}, this.state.args, { defaultValue: null });
-        this.setState({
-          args,
-          onError: 'default-value',
-          errorStrategy,
-        });
-        break;
-      }
-      case 'default-value':
-      case 'fail':
-      case 'delete-row': {
-        this.setState({
-          onError: errorStrategy,
-          errorStrategy,
-        });
-        break;
-      }
-      default:
-        throw new Error(`Unkown error strategy ${errorStrategy}`);
-    }
+    const transformation = this.state.transformation;
+    const defaultValue = errorStrategy === 'empty-cell' ?
+      null : transformation.getIn(['args', 'defaultValue']);
+    this.setState({
+      errorStrategy,
```

Comment:
Transformations always default to "blank cell" on error, no matter what is chosen in the "if cell format is invalid" input

I think this is because we record the new error strategy in `this.state.errorStrategy`, but in `onApply` we only pass `this.state.transformation`, which always has an `onError` value of `default-value`